.github/workflows/default.yml

```diff
@@ -0,0 +1,75 @@
+name: Build Project
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, windows-latest ]
+    name: '[${{ matrix.os }}] Build plugin'
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: set up Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: 16
+
+      - name: Cache .gradle
+        uses: burrunan/gradle-cache-action@v1
+        with:
+          gradle-distribution-sha-256-sum-warning: false
+
+      - name: Gradle Wrapper Validation
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: ./gradlew assemble
+        uses: eskatos/gradle-command-action@v1
+        with:
+          arguments: assemble
+
+      - name: ./gradlew check
+        uses: eskatos/gradle-command-action@v1
+        with:
+          arguments: check
+
+      - name: Upload test results
+        if: ${{ always() }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: test-results-${{ matrix.os }}
+          path: "${{ github.workspace }}/**/build/reports/tests"
```

Comment:
This will make test reports easily available under `artifacts` tab:
![image](https://user-images.githubusercontent.com/36954793/116826131-ca261380-ab92-11eb-941c-c33a57df6eac.png)
They weren't available on travis, which made it a bit problematic to investigatie test failures there